### DISCUSSION TITLE
Section 5, Lesson 20: Why not just use the simplified invariant?

### DISCUSSION THREAD

**Discussion by antoniopgs** [Cyfrin/security-and-auditing-full-course-s23#58] [Q&A] [answered]
Pat says:
"It's hard to write an assert for `x * y = k`"

But is it?
Why not just do `assert(tSwap.balanceOf(x) * tSwap.balanceOf(y) == k)`?

Where:
`k = pool initialBalance of x * pool initialBalance of y`

**Comment by TilakMaddy** [comment, marked as answer]
Hmm nope! I actually had the same question but this is the conclusion I have come to https://github.com/Cyfrin/security-and-auditing-full-course-s23/discussions/49
